### PR TITLE
feat(events): get events list is ready for test

### PR DESCRIPTION
**Pull request by DiLiUA**
- reworked methods: getNonLiveEventsAmountData, getEventsDataByQuery, saveNewEvent

### CODE REVIEW

File: routes/event.ts [hunk 1]
```diff
@@ -30,16 +31,33 @@ function getNonLiveEventsAmountData(req: Request, res: Response): void {
...
+    })
+    .catch(err => {
+      const status = 500;
+      res.status(status).send({message: err});
+    });
 }
 
 function getEventsDataByQuery(req: Request, res: Response): void {
   const query = req.query;
+  const limit: number = query.limit ? +query.limit : null;
```
Comment (Ugmaxie) [inline routes/event.ts]: Types... 
number | null

File: routes/event.ts [hunk 2]
```diff
@@ -72,8 +90,10 @@ function getEventsDataByQuery(req: Request, res: Response): void {
 
   events
     .find(commonQuery, {buyers: false})
+    .limit(limit)
     .lean(true)
-    .exec((err: Error, data: Event[]): void => {
+    .exec()
+    .then((data: Event[]) => {
```
Comment (Ugmaxie) [inline routes/event.ts:96]: data =>

File: routes/event.ts [hunk 3]
```diff
@@ -99,22 +123,26 @@ function saveNewEvent(req: Request, res: Response): void {
...
-        $push: {
-          'shows.owned': event._id
-        }
-      })
-      .exec((error: Error, data: UpdatedDb): void => {
-        res.json({success: !err, data: {message: 'user saved', newId: event._id}, error: err});
-      });
-  });
+  newEvent.save()
+    .then((event: Event) => {
```
Comment (Ugmaxie) [inline routes/event.ts]: event =>

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: routes/event.ts (hunk 2 above)
Comment (DiLiUA) [reply routes/event.ts:96]: If I delete type I have tslint Error `Property 'map' does not exist on type 'Object'`

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**